assimilation_code/modules/assimilation/filter_mod.f90

```diff
@@ -1761,6 +1765,56 @@ end subroutine obs_space_diagnostics
 
 !-------------------------------------------------------------------------
 
+subroutine obs_space_sync_QCs(obs_fwd_op_ens_handle,  &
+   seq, keys, num_obs_in_set, OBS_GLOBAL_QC_COPY, DART_qc_index)
+
+! If QCs were updated in the assimilation loop but posterior forward operators
+! are not being computed, collect any updated QCs into the output obs_seq.
+
+type(ensemble_type),     intent(inout) :: obs_fwd_op_ens_handle
+integer,                 intent(in)    :: num_obs_in_set
+integer,                 intent(in)    :: keys(num_obs_in_set)
+type(obs_sequence_type), intent(inout) :: seq
+integer,                 intent(in)    :: OBS_GLOBAL_QC_COPY
+integer,                 intent(in)    :: DART_qc_index
+
+integer               :: j
+integer               :: io_task, my_task
+real(r8), allocatable :: obs_temp(:)
+real(r8)              :: rvalue(1)
+
+! this is a query routine to return which task has 
+! logical processing element 0 in this ensemble.
+io_task = map_pe_to_task(obs_fwd_op_ens_handle, 0)
+my_task = my_task_id()
+
+! Make var complete for get_copy() calls below.
```

Comment:
This is for readability:
Rather than have the transpose + a note about the transpose is needed 'below' you can  move the transpose to line 1804.
That way the transpose is next to the call that requires you to be var complete.

``
call all_copies_to_all_vars
call get_copy
``
```suggestion
```

Reply:
Committed these edits.